In [1]:
import pandas as pd
import numpy as np
from matplotlib import pyplot as plt
import seaborn as sns

%matplotlib inline

5. Using the pandas read_csv() function, read the GDP dataset into your notebook as a DataFrame called gdp_df.

    Take a look at the first 10 rows.  
    Look at the last 5 rows. Do you see a problem?  
    Redo the read_csv() call to correct this issue - do not modify the original csv file.  



In [12]:
gdp_df = pd.read_csv('../data/gdp_per_capita.csv',
                     nrows = 6868
                    )

In [13]:
gdp_df.head(10)

Country or Area  Year        Value  Value Footnotes
0     Afghanistan  2019  2065.036398              NaN
1     Afghanistan  2018  2033.779002              NaN
2     Afghanistan  2017  2058.383832              NaN
3     Afghanistan  2016  2057.062164              NaN
4     Afghanistan  2015  2068.265904              NaN
5     Afghanistan  2014  2102.385234              NaN
6     Afghanistan  2013  2116.465454              NaN
7     Afghanistan  2012  2075.491747              NaN
8     Afghanistan  2011  1904.559799              NaN
9     Afghanistan  2010  1957.029338              NaN

In [14]:
gdp_df.tail()

Country or Area  Year        Value  Value Footnotes
6863        Zimbabwe  1994  3263.934978              NaN
6864        Zimbabwe  1993  3033.504852              NaN
6865        Zimbabwe  1992  3054.889178              NaN
6866        Zimbabwe  1991  3426.598094              NaN
6867        Zimbabwe  1990  3324.348171              NaN

6. Drop the 'Value Footnotes' column, and rename the remaining columns to ‘Country’, ‘Year’, and ‘GDP_Per_Capita’.

In [16]:
gdp_df = gdp_df.drop('Value Footnotes',
                    axis = 1
                    )

In [20]:
gdp_df = gdp_df.rename(columns = {'Country or Area': 'Country',
                                  'Value': 'GDP_Per_Capita'
                                 })

In [21]:
gdp_df

Country  Year  GDP_Per_Capita
0     Afghanistan  2019     2065.036398
1     Afghanistan  2018     2033.779002
2     Afghanistan  2017     2058.383832
3     Afghanistan  2016     2057.062164
4     Afghanistan  2015     2068.265904
...           ...   ...             ...
6863     Zimbabwe  1994     3263.934978
6864     Zimbabwe  1993     3033.504852
6865     Zimbabwe  1992     3054.889178
6866     Zimbabwe  1991     3426.598094
6867     Zimbabwe  1990     3324.348171

[6868 rows x 3 columns]

7. How many rows and columns does gdp_df have? What are the data types of its columns? If any of the columns are not the expected types, figure out why and fix it.

In [23]:
gdp_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 6868 entries, 0 to 6867
Data columns (total 3 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   Country         6868 non-null   object 
 1   Year            6868 non-null   int64  
 2   GDP_Per_Capita  6868 non-null   float64
dtypes: float64(1), int64(1), object(1)
memory usage: 161.1+ KB


8. Which years are represented in this dataset? Take a look at the number of observations per year. What do you notice?

In [24]:
gdp_df.value_counts('Year')

Year
2017    238
2016    238
2015    238
2014    238
2013    238
2011    237
2018    237
2012    237
2009    236
2010    236
2006    235
2007    235
2008    235
2004    234
2005    234
2002    233
2003    233
2001    232
2019    231
2000    231
1999    224
1998    223
1997    223
1996    221
1995    221
1994    214
1993    212
1992    209
1991    208
1990    207
dtype: int64

The years 1990-2019 are included in the data. The number of observations increased from 207 to 238 over the timespan, until 2017 when it began decreasing.

9. How many countries are represented in this dataset? Which countries are least represented in the dataset? Why do you think these countries have so few observations?

In [25]:
gdp_df.value_counts('Country')

Country
Lebanon                      30
Middle income                30
Morocco                      30
Mozambique                   30
Myanmar                      30
                             ..
Nauru                        16
Cayman Islands               13
Sint Maarten (Dutch part)    10
Turks and Caicos Islands      9
Djibouti                      7
Length: 238, dtype: int64

In [26]:
gdp_df[gdp_df['Country'] == 'Djibouti']

Country  Year  GDP_Per_Capita
1477  Djibouti  2019     5534.749116
1478  Djibouti  2018     5214.088499
1479  Djibouti  2017     4885.220563
1480  Djibouti  2016     4709.549234
1481  Djibouti  2015     4488.827214
1482  Djibouti  2014     4238.996239
1483  Djibouti  2013     4028.446052

There are 238 countries in the data. Nauru, The Cayman Islands, Sint Maarten, Turks and Caicos Islands, and Djibouti are the least represented in the data. 

10. Create a new dataframe by subsetting gdp_df to just the year 2014. Call this new dataframe gdp_2014.

In [27]:
gdp_2014 = gdp_df[gdp_df['Year'] == 2014]

In [28]:
gdp_2014.head()

Country  Year  GDP_Per_Capita
5            Afghanistan  2014     2102.385234
23               Albania  2014    11623.866679
53               Algeria  2014    11512.705405
83                Angola  2014     8239.828837
113  Antigua and Barbuda  2014    18104.241470